src/newmagic.cpp

```diff
@@ -663,11 +663,38 @@ bool conjuring_drain(struct char_data *ch, int force)
   else
     drain = SERIOUS;
   drain = convert_damage(stage(-success_test(GET_CHA(ch), force), drain));
-  if (force > GET_MAG(ch) / 100)
+  
+  // Physical drain.
+  if (force > GET_MAG(ch) / 100) {
+    // Iterate through bioware to find relevant bioware.
+    for (struct obj_data *bio = ch->bioware; bio && drain > 0; bio = bio->next_content) {
+      if (GET_OBJ_VAL(bio, 0) == BIO_PLATELETFACTORY && drain >= 3)
```

Comment:
This will produce inconsistent results depending on the ordering of the player's bioware list. If they're about to take 3 drain, and the trauma damper comes before the platelet factory, they won't get the effects of both. Maybe make a new `delta` variable to tally up the total expected changes, then apply them all at once after the loop?

Also, `GET_BIOWARE_TYPE()` here please!

Reply:
Hmm. You're right, I didn't think of that. You know this is exactly how it's implemented in spell_drain as well. So both need fixing there.